src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -154,24 +156,63 @@ public AzureAppConfigurationOptions UseFeatureFlags(Action<FeatureFlagOptions> c
                     string.Format(ErrorMessages.CacheExpirationTimeTooShort, MinimumFeatureFlagsCacheExpirationInterval.TotalMilliseconds));
             }
 
-            if (!(_kvSelectors.Any(selector => selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker) && selector.LabelFilter.Equals(options.Label))))
+            if (options.FeatureFlagSelectors.Count() != 0 && options.Label != null)
             {
-                Select(FeatureManagementConstants.FeatureFlagMarker + "*", options.Label);
+                throw new InvalidOperationException($"Please select feature flags by either the {nameof(options.Select)} method or by setting the {nameof(options.Label)}, not both.");
+            }
+            
+            if (options.FeatureFlagSelectors.Count() == 0 && options.Label == null)
+            {
+                // No Select clause and the Label property is not set
+                options.FeatureFlagSelectors.Add(new KeyValueSelector
+                {
+                    KeyFilter = FeatureManagementConstants.FeatureFlagMarker + "*",
+                    LabelFilter = LabelFilter.Null
+                });  
+            }
+            else if (options.FeatureFlagSelectors.Count() == 0 && options.Label != null)
+            {
+                // No Select clause but the Label property is set
+                options.FeatureFlagSelectors.Add(new KeyValueSelector
+                {
+                    KeyFilter = FeatureManagementConstants.FeatureFlagMarker + "*",
+                    LabelFilter = options.Label
+                });
             }
 
+            _featureFlagPrefixes.AddRange(options.FeatureFlagPrefixes);
```

Comment:
>_featureFlagPrefixes.AddRange(options.FeatureFlagPrefixes); [](start = 12, length = 59)

Are we preventing duplicates?

Reply:
Yes, the underlying data structure is a SortedSet.

Reply:
I see this line
`        private List<string> _featureFlagPrefixes = new List<string>();`
Where's the sorted set?

Reply:
The logic to process user input and add to a list of feature flag prefixes is in `FeatureFlagOptions` [here](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/234/files#diff-6481f817620bb03ea907e90b07f2bdea0493301347d8adb2c6b3af4f6ac45c4dR16)

Reply:
I found it after moving files. But it looks like the duplication is still possible through multiple usefeatureflags invocations.

Reply:
You're right. `AzureAppConfigurationOptions` should have the SortedSet in that case.